CONTRIBUTING.md

```diff
@@ -0,0 +1,197 @@
+- `CONTAINER_REGISTRY_USERNAME` & `CONTAINER_REGISTRY_PASSWORD`: Authentication information for the the repo above.
+
+### Development environment on minikube for the traceloop gadget
+
+It's possible to make changes to traceloop and test them on minikube locally without pushing container images to any registry.
+
+* Make sure the git repositories `traceloop` and `inspektor-gadget` are cloned in sibling directories
+* Minikube with the Docker driver does not work for traceloop. You can use another driver, for example:
+```
+$ minikube start --driver=kvm2
+```
+* Install Inspektor Gadget on minikube as usual:
+```
+$ kubectl gadget deploy | kubectl apply -f -
+```
+* Make changes in the traceloop repository and compile with `make`
+* Generate the new gadget image and deploy it to minikube:
+```
+$ make -C gadget-container/ minikube
+```
+
+Note that the minikube image only works with the traceloop gadget.
+
+## Testing
+
+### Unit tests
+
+You can run the different unit tests by:
+
+```
+make test
+```
+
+### Integration tests
+
+The integration tests use a Kubernetes cluster to deploy and test Inspektor Gadget.
+Be sure that you have a valid kubeconfig and run:
+
+```
+export KUBECONFIG=... # not needed if valid config in $HOME/.kube/config
+make integration-tests
+```
+
+## Authoring PRs
+
+These are general guidelines for making PRs/commits easier to review:
+
+ * Commits should be atomic and self-contained. Divide logically separate changes
+   to separate commits. This principle is best explained in the the Linux Kernel
+   [submitting patches][linux-sep-changes] guide.
+
+ * Commit messages should explain the intention, _why_ something is done. This,
+   too, is best explained in [this section][linux-desc-changes] from the Linux
+   Kernel patch submission guide.
+
+ * Commit titles (the first line in a commit) should be meaningful and describe
+   _what_ the commit does.
+
+ * Don't add code you will change in a later commit (it makes it pointless to
+   review that commit), nor create a commit to add code an earlier commit should
+   have added. Consider squashing the relevant commits instead.
+
+ * It's not important to retain your development history when contributing a
+   change. Use `git rebase` to squash and order commits in a way that makes them easy to
+   review. Keep the final, well-structured commits and not your development history
+   that led to the final state.
+
+ * Consider reviewing the changes yourself before opening a PR. It is likely
+   you will catch errors when looking at your code critically and thus save the
+   reviewers (and yourself) time.
+
+ * Use the PR's description as a "cover letter" and give the context you think
+   reviewers might need. Use the PR's description to explain why you are
+   proposing the change, give an overview, raise questions about yet-unresolved
+   issues in your PR, list TODO items etc.
+
+PRs which follow these rules are easier to review and merge.
```

Comment:
We really need to add this stuff to the `contribute` repo. Just a comment to say we may change this part later and just point to the contribute repo.